### DISCUSSION TITLE
Concurrent 'buildah bud ...' on same host?

### DISCUSSION THREAD

**Discussion by jonstelly** [containers/buildah#3400] [Q&A] [answered]
### Two questions:

1) Is concurrent building with `buildah bud ...` on a single host supported?
2) If it is, is there something I'm doing wrong?

### Explanation

I've got some images that take a bit of time to build, I'm moving from `docker buildx` and the way that buildx built different platforms in parallel was helpful.

I tried moving to `buildah bud` using background tasks, one command per platform but I seem to be getting all sorts of random errors that make it sound like the commands are stepping on each other and the local image cache.  FWIW, this is on Ubuntu 20.04 and using qemu-user and qemu-static to handle multiple binary formats.

```
buildah bud --platform linux/amd64 --pull --build-arg=TARGETARCH=amd64 --build-arg=TARGETOS=linux \
 --build-arg=TARGETPLATFORM=linux/amd64 --tag localhost/image:latest-amd64 .

buildah bud --platform linux/arm64 --pull --build-arg=TARGETARCH=arm64 --build-arg=TARGETOS=linux \
 --build-arg=TARGETPLATFORM=linux/arm64 --tag localhost/image:latest-arm64 .
```

If I run those commands sequentially they run as expected.  If I try to run them both at the same time I get all sorts of errors.  I'm being careful about tagging each concurrent build with a different tag.  None of them are `FROM scratch`, each `FROM ...` line would be to like `ubuntu:rolling` which should be a manifest so each build step/layer should start from a different underlying layer.  So it seems to me like they shouldn't really be having layer conflicts.

If this is supported I'll capture an assortment of the errors and create an issue.

**Comment by vrothberg** [comment, marked as answer]
Thank you for reaching out, @jonstelly!

Yes, please open an issue, ideally with a reproducer.

**Reply by jonstelly** [reply]
This issue had happened a few months ago.  I just went and tested again, ran quite a few iterations of concurrent builds and everything seems to be working correctly now.  No need for an issue, thanks!
